packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -76,6 +76,16 @@ abstract mixin class WidgetsBindingObserver {
   /// {@macro flutter.widgets.AndroidPredictiveBack}
   Future<bool> didPopRoute() => Future<bool>.value(false);
 
+  Future<bool> startBackGesture(AndroidBackEvent backEvent) =>
```

Comment:
Generally I worry about these classes that interact with platform channels getting too big, so I think these should be split out into their own class that is specific to predictive back gestures.

I'm remembering that I mentioned this on the engine PR too (https://github.com/flutter/engine/pull/49093#discussion_r1479028490). I would say yes, you should create an entirely separate channel for this.

Reply:
Looks good now with the `backGesture` channel, thanks.